src/service_info.rs

```diff
@@ -499,6 +499,20 @@ impl TxtProperties {
     pub fn get_property_val_str(&self, key: &str) -> Option<&str> {
         self.get(key).map(|x| x.val_str())
     }
+
+    /// Returns a map of properties, where the key is the property key.
+    pub fn into_property_map_str(self) -> HashMap<String, String> {
+        self.properties
+            .into_iter()
+            .filter_map(|p| {
+                // Skip the property if the value is non-empty and not valid UTF-8.
```

Comment:
The comment seems wrong. We should skip if the value is **empty** and not valid UTF-8

Reply:
I indeed meant **non-empty**.

As  TXT record allows empty value for a key (see [RFC](https://datatracker.ietf.org/doc/html/rfc6763#section-6.4), I wanted to do:

- If the value is empty, this function sets an empty string for this attribute.
- If the value is non-empty but fails to convert to UTF-8, this function treats it as invalid, and skip this attribute. (Maybe we should log a debug message).

Reply:
I did not know that, clear now, thank you for your explanation! :)